### PR TITLE
Upgrade to Ember v3.20, drop Node 8

### PR DESCRIPTION
**Pull request by raido**
This is prerequisite for this PR: https://github.com/adopted-ember-addons/ember-page-title/pull/168

Closes: https://github.com/adopted-ember-addons/ember-page-title/issues/169

### CODE REVIEW

File: .template-lintrc.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 'use strict';
 
 module.exports = {
-  extends: 'recommended'
+  extends: 'recommended' // TODO migrate to "octane"
```
Comment (raido) [inline .template-lintrc.js:4]: This should be done in separate PR, there's a lot of template issues.

File: package.json [hunk 2]
```diff
@@ -16,52 +16,61 @@
...
+    "test:ember": "ember test",
+    "test:ember-compatibility": "ember try:each",
     "release": "standard-version"
   },
   "dependencies": {
-    "ember-cli-babel": "^7.7.3",
+    "ember-cli-babel": "^7.21.0",
     "ember-cli-head": "^1.0.0",
-    "ember-cli-htmlbars": "^3.0.1",
+    "ember-cli-htmlbars": "^5.2.0",
```
Comment (rwjblue) [inline package.json:32]: Do we actually have any templates in addon/? If not, can you move this to devDeps?

File: tests/acceptance/posts-test.js [hunk 3]
```diff
@@ -73,15 +73,15 @@ module('Acceptance: title', function(hooks) {
     await visit('/feeds/tomster');
     assert.equal(title(), 'Tomster (@tomster)');
 
-    await click('#zoey');
+    click('#zoey');
```
Comment (rwjblue) [inline tests/acceptance/posts-test.js:76]: This seems odd, why remove the await here?

File: tests/acceptance/posts-test.js [hunk 4]
```diff
@@ -73,15 +73,15 @@ module('Acceptance: title', function(hooks) {
...
     await waitUntil(() => {
-      return find('div').innerText !== 'Loading...';
+      return !find('div[data-test-substate-loading]');
```
Comment (rwjblue) [inline tests/acceptance/posts-test.js:78]: This find probably needs to be imported

File: tests/acceptance/posts-test.js [hunk 5]
```diff
@@ -73,15 +73,15 @@ module('Acceptance: title', function(hooks) {
...
     });
     assert.equal(title(), 'Zoey (@zoey)');
 
     await click('#tomster');
     await waitUntil(() => {
-      return find('div').innerText !== 'Loading...';
+      return !find('div[data-test-substate-loading]');
```
Comment (rwjblue) [inline tests/acceptance/posts-test.js:84]: Same here, I think if you import you don't need the bang

File: tests/dummy/app/index/template.hbs [hunk 6]
```diff
@@ -38,7 +38,7 @@
             </code>
           {{/p.matryoshka}}
 
-          <button onclick={{action "add" model.tokens}}>Add title</button>
+          <button type="button" onclick={{action "add" model.tokens}}>Add title</button>
```
Comment (rwjblue) [inline tests/dummy/app/index/template.hbs:41]: Let's use `on` here instead of `onclick`

File: tests/acceptance/posts-test.js (hunk 3 above)
Comment (raido) [reply tests/acceptance/posts-test.js:76]: Ah, sorry this is typo from debugging behaviour in DOM elements used for selectors below.

File: tests/acceptance/posts-test.js (hunk 4 above)
Comment (raido) [reply tests/acceptance/posts-test.js:78]: find helper is imported, I'll use === null expression instead to give better intent what is meant with waiting element to disappear.

File: tests/dummy/app/index/template.hbs (hunk 6 above)
Comment (raido) [reply tests/dummy/app/index/template.hbs:41]: This would be updated in next PR together all other template errors, there's hundreds of them.

File: package.json (hunk 2 above)
Comment (raido) [reply package.json:32]: It's here because of ember-cli-head -> head.hbs in app/templates folder.

File: tests/acceptance/posts-test.js (hunk 3 above)
Comment (raido) [reply tests/acceptance/posts-test.js]: Fixed.

File: package.json (hunk 2 above)
Comment (rwjblue) [reply package.json:32]: Gotcha. So the other PR will remove that template and move this to devDeps?

File: tests/acceptance/posts-test.js (hunk 4 above)
Comment (rwjblue) [reply tests/acceptance/posts-test.js]: Doh! I misread that as the TypeScript syntax 🤦‍♂️.

File: tests/acceptance/posts-test.js (hunk 5 above)
Comment (rwjblue) [reply tests/acceptance/posts-test.js]: (Obviously mistaken here, sorry)

File: tests/dummy/app/index/template.hbs (hunk 6 above)
Comment (rwjblue) [reply tests/dummy/app/index/template.hbs:41]: Ah, kk. 👍

**Comment by rwjblue** [comment]
Looks good to me! Thanks @raido!

File: package.json (hunk 2 above)
Comment (raido) [reply package.json:32]: If we get this sorted out: https://github.com/adopted-ember-addons/ember-page-title/pull/168 - then yes.
